src/scripts/index.js

```diff
@@ -0,0 +1,333 @@
+/**
+ * Класс, объекты которого описывают параметры продукта.
+ *
+ * @constructor
+ * @param size          Размер
+ * @param stuff         Начинка
+ */
+function Product(size, stuff) {
+    this.size = size;
+    this.stuff = stuff;
+}
+
+/**
+ * Получить размер
+ * @returns {String}    Размер
+ */
+Product.prototype.getSize = function () {
+    return this.size;
+};
+
+/**
+ * Получить начинку
+ * @returns {String}    Начинка
+ */
+Product.prototype.getStuff = function () {
+    return this.stuff;
+};
+
+/**
+ * Изменить размер
+ * @param size          Размер
+ */
+Product.prototype.setSize = function (size) {
+    this.size = size;
+};
+
+/**
+ * Изменить начинку
+ * @param stuff         Начинка
+ */
+Product.prototype.setStuff = function (stuff) {
+    this.stuff = stuff;
+};
+
+/**
+ * Узнать стоимость продукта
+ * @returns {Number}    Стоимость продукта
+ */
+Product.prototype.calculatePrice = function () {
+    var price = 0;
+
+    price += this.stuff.price;
+    if (this.size.coefficient) {
+        price *= this.size.coefficient;
+    } else {
+        price += this.size.price;
+    }
+    return price;
+};
+
+/**
+ * Узнать калорийность продукта
+ * @returns {Number}    Калорийность продукта
+ */
+Product.prototype.calculateCalories = function () {
+    var calories = 0;
+
+    calories += this.stuff.calories;
+    if (this.size.coefficient) {
+        calories *= this.size.coefficient;
+    } else {
+        calories += this.size.calories;
+    }
+    return calories;
+};
+
+/**
+ * Класс, объекты которого описывают параметры гамбургера.
+ * Наследуется от класса Product
+ *
+ * @constructor
+ * @param size     Размер
+ * @param stuff    Начинка
+ */
+function Hamburger(size, stuff) {
+    Product.call(this, size, stuff);
+    this.name = 'hamburger';
+}
+
+Hamburger.prototype = Object.create(Product.prototype);
+
+/* Размеры, виды начинок и добавок */
+Hamburger.prototype.SIZE_SMALL = { name: "Small", price: 50, calories: 20 };
+Hamburger.prototype.SIZE_LARGE = { name: "Large", price: 100, calories: 40 };
+Hamburger.prototype.STUFF_CHEESE = { name: "cheese", price: 10, calories: 20 };
+Hamburger.prototype.STUFF_SALAD = { name: "salad", price: 20, calories: 5 };
+Hamburger.prototype.STUFF_POTATO = { name: "potato", price: 15, calories: 10 };
```

Comment:
@TheUnforgivenOne It's better to place static constants as function's properties, like `Hamburger.SIZE_SMALL` as it listed in the task